pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			ctx.Printlnf("\nERROR: The given deployment '%s' wasn't found.", deploymentName)
-			return printExistingDeployments(ctx, cl, ns)
-		}
 		return err
 	}
-	ctx.Println("The deployment was scaled to 0")
-	if err := scaleBack(ctx, cl, namespacedName, originalReplicas); err != nil {
-		ctx.Printlnf("Scaling the deployment '%s' in namespace '%s' back to '%d' replicas wasn't successful", originalReplicas)
-		ctx.Println("Please, try to contact administrators to scale the deployment back manually")
-		return err
+	//if there is no operator deployment, no need for restart
+	if len(operatorDeploymentList.Items) == 0 {
+		return fmt.Errorf("no operator based deployment found in namespace %s , hence no restart happened", ns)
 	}
+	//Deleting the pods of the operator based deployment  and then checking the status
+	for _, operatorDeployment := range operatorDeploymentList.Items {
+		ctx.Printlnf("Proceeding to delete the Pods of %v", operatorDeployment.Name)
 
-	ctx.Printlnf("The deployment was scaled back to '%d'", originalReplicas)
-	return nil
-}
+		if err := deleteAndWaitForPods(ctx, cl, operatorDeployment); err != nil {
+			return err
+		}
 
-func restartHostOperator(ctx *clicontext.CommandContext, hostClient runtimeclient.Client, hostNamespace string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := hostClient.List(context.TODO(), deployments,
-		runtimeclient.InNamespace(hostNamespace),
-		runtimeclient.MatchingLabels{"olm.owner.namespace": "toolchain-host-operator"}); err != nil {
-		return err
+		ctx.Printlnf("Checking the status of the deleted pod's deployment %v", operatorDeployment.Name)
+		//check the rollout status
+		if err := checker(ctx, f, ioStreams, "kubesaw-control-plane=kubesaw-controller-manager"); err != nil {
+			return err
+		}
+	}
+
+	if len(nonOperatorDeploymentList.Items) == 0 {
+		// if there are no non-operator deployments
+		ctx.Printlnf("No Non-operator deployment found in namespace %s, hence no restart happened", ns)
+		return nil
 	}
-	if len(deployments.Items) != 1 {
-		return fmt.Errorf("there should be a single deployment matching the label olm.owner.namespace=toolchain-host-operator in %s ns, but %d was found. "+
-			"It's not possible to restart the Host Operator deployment", hostNamespace, len(deployments.Items))
+	for _, nonOperatorDeployment := range nonOperatorDeploymentList.Items {
+		if nonOperatorDeployment.Name != "autoscaling-buffer" {
+			ctx.Printlnf("Proceeding to restart the non-operator deployment %v", nonOperatorDeployment.Name)
+
+			if err := restarter(ctx, nonOperatorDeployment, f, ioStreams); err != nil {
+				return err
+			}
+			//check the rollout status
+			ctx.Printlnf("Checking the status of the rolled out deployment %v", nonOperatorDeployment.Name)
+			if err := checker(ctx, f, ioStreams, "toolchain.dev.openshift.com/provider=codeready-toolchain"); err != nil {
+				return err
+			}
+			return nil
```

Comment:
wouldn't this break the loop so only one deployment would be restarted? let's verify this also in the tests that it restarts more than one non-OLM based deployments

Reply:
i dont understand, there will be only 1 non-OLM based deployment for host (reg service) and member(web-hook).? or are we expecting more which we need to restart

Reply:
ah, ok. it wasn't clear from the code. Then either add a comment there, or just simply drop the return so it can automatically support any number of deployments. 
It's also not consistent with the way you iterate over OLM-based deployments

Reply:
updated it with comments